Cargo.toml

```diff
@@ -38,6 +38,10 @@ bitcoinconsensus = { version = "0.19.0-3", optional = true }
 # Needed by bdk_blockchain_tests macro
 core-rpc = { version = "0.14", optional = true }
 
+# Platform-specific dependencies
+[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
+tokio = { version = "1", features = ["rt"] }
```

Comment:
Please correct me if I'm wrong, but this PR re-introduces a hard dependency on `tokio` just so we can test `esplora-reqwest` feature, that seems like a bad choice to me. A user of `bdk` that wishes to use `reqwest` will always be using `async`, why do we want to introduce a hard dependency on `tokio` just to be able to test in a manner that the library will not be used?

The `esplora-reqwest` feature can be tested coupled with either `async-interface` feature or using WASM target.